numpy/random/mtrand.pyx

```diff
@@ -4112,21 +4113,20 @@ cdef class RandomState:
             string.memcpy(data + j * stride, data + i * stride, itemsize)
             string.memcpy(data + i * stride, buf, itemsize)
 
-    def permutation(self, object x):
+    def permutation(self, object x, axis=0):
```

Comment:
This file is frozen and cannot be changed except for issues relating to compilation, not features or bugs.  These changes need to be removed.

Reply:
https://github.com/numpy/numpy/pull/13838#pullrequestreview-254540116 suggests doing a documentation-cleaning pass on `numpy.random` before 1.17 is cut. Perhaps that would be a good time to add a comment at the top of `mtrand.pyx` to forewarn future contributors from adding features to the legacy code.

Reply:
Agree. It should be clear that NEP-19 has frozen this file.

Reply:
Oh, I see. If I cannot add features to `mtrand.pyx`, can you please refer which file should I add features to?

Reply:
New features can be implemented for Generator in generator.pyx.  They cannot be added to RandomState (in mtrand.pyx).

Reply:
Ok, then I will only keep the changes in `generator.pyx`, and also add similar feature to `shuffle`.